File: Exo2.2.py
import os

pid1 = os.fork()  # Création du fils 1

if pid1 == 0:
    print("Je suis le fils 1")
    os._exit(0)  # Terminer le fils 1

pid2 = os.fork()  # Création du fils 2

if pid2 == 0:
    print("Je suis le fils 2")
    os._exit(0)  # Terminer le fils 2

# Processus père
print("Je suis le père")

# Attendre que les deux fils terminent
os.waitpid(pid1, 0)
os.waitpid(pid2, 0)

File: Exo2.2.c
#include <stdio.h>
#include <unistd.h>
#include <sys/types.h>
#include <sys/wait.h>

int main()
{
    pid_t pid1, pid2;

    pid1 = fork();  // Création du fils 1
    if (pid1 < 0)
    {
        perror("Erreur de création de fork");
        return (1);
    } 
    else if (pid1 == 0)
    {
        // Processus fils 1
        printf("Je suis le fils 1\n");
        return (0);
    }

    // Création du fils 2
    pid2 = fork();
    if (pid2 < 0) 
    {
        
        perror("Erreur de création de fork");
        return (1);
    } 
    else if (pid2 == 0)
    {
        // Processus fils 2
        printf("Je suis le fils 2\n");
        return (0);
    }

    // Processus père
    printf("Je suis le père\n");

    // Attendre que les deux fils se terminent
    wait(NULL);
    wait(NULL);

    return (0);
}